File: solutions/python/bob/3/bob.py
def response(hey_bob: str) -> str:
    is_question = hey_bob.strip().endswith('?')
    is_all_capital = hey_bob.isupper()
    is_silent = not hey_bob.strip()

    if is_question and is_all_capital:
        return 'Calm down, I know what I\'m doing!'
    if is_question:
        return 'Sure.'
    if is_silent:
        return 'Fine. Be that way!'
    if is_all_capital:
        return 'Whoa, chill out!'
    return 'Whatever.'

File: solutions/ruby/bob/1/bob.rb
module Bob
  RESPONSE_TO_QUESTION = 'Sure.'
  RESPONSE_TO_YELL = 'Whoa, chill out!'
  RESPONSE_TO_YELLED_QUESTION = 'Calm down, I know what I\'m doing!'
  RESPONSE_TO_SILENT = 'Fine. Be that way!'
  DEFAULT_RESPONSE = 'Whatever.'


  def self.hey(remark)
    has_letter = remark =~ /[A-Za-z]/
    is_question = remark =~ /\?[\s]*\z/
    is_all_captial = remark =~ /^[A-Z\W\d]+\z/
    is_silent = remark =~ /^\s*\z/

    if is_silent
      RESPONSE_TO_SILENT
    elsif is_question and is_all_captial and has_letter
      RESPONSE_TO_YELLED_QUESTION
    elsif is_question
      RESPONSE_TO_QUESTION
    elsif is_all_captial and has_letter
      RESPONSE_TO_YELL
    else
      DEFAULT_RESPONSE
    end
  end
end